qiskit_experiments/library/randomized_benchmarking/layer_fidelity_unitary.py

```diff
@@ -0,0 +1,606 @@
+            two_qubit_basis_gates: Optional, 2q-gates to use for transpiling the inverse.
+                            If not specified (but ``backend`` is supplied),
+                            all 2q-gates supported in the backend are automatically set.
+            one_qubit_basis_gates: Optional, 1q-gates to use for implementing 1q-Clifford operations.
+                            If not specified (but ``backend`` is supplied),
+                            all 1q-gates supported in the backend are automatically set.
+            layer_barrier (bool): Optional, enforce a barrier across the whole layer.
+                Default is True, which is the defined protocol for layer fidelity.
+                If this is set to false the code runs
+                simultaneous direct 1+2Q RB without a barrier across all qubits.
+            min_delay: Optional. Define a minimum delay in each 2Q layer in units of dt. This
+                delay operation will be applied in any 1Q edge of the layer during the 2Q gate layer
+                in order to enforce a minimum duration of the 2Q layer. This enables some crosstalk
+                testing by removing a gate from the layer without changing the layer duration. If not
+                None then is a list equal in length to the number of two_qubit_layers.  Note that
+                this options requires at least one 1Q edge (a qubit in physical_qubits but
+                not in two_qubit_layers) to be applied. Also will not have an impact on the 2Q gates
+                if layer_barrier=False.
+
+        Raises:
+            QiskitError: If any invalid argument is supplied.
+        """
+        # Compute full layers
+        full_layers = []
+        for two_q_layer in two_qubit_layers:
+            qubits_in_layer = {q for qpair in two_q_layer for q in qpair}
+            if len(qubits_in_layer) != 2 * len(two_q_layer):
+                raise QiskitError("two_qubit_layers have a layer with gates on non-disjoint qubits")
+            for q in qubits_in_layer:
+                if q not in physical_qubits:
+                    raise QiskitError(f"Qubit {q} in two_qubit_layers is not in physical_qubits")
+            layer = two_q_layer + [(q,) for q in physical_qubits if q not in qubits_in_layer]
+            full_layers.append(layer)
+
+        # Initialize base experiment
+        super().__init__(
+            physical_qubits, analysis=LayerFidelityAnalysis(full_layers), backend=backend
+        )
+        # assert isinstance(backend, BackendV2)
+
+        # Verify parameters
+        if len(set(lengths)) != len(lengths):
+            raise QiskitError(f"The lengths list {lengths} should not contain duplicate elements.")
+        if num_samples <= 0:
+            raise QiskitError(f"The number of samples {num_samples} should be positive.")
+
+        if two_qubit_gates is None:
+            raise QiskitError("Must specify a set of two qubit gate circuits.")
```

Comment:
Can we just move `two_qubit_gates` up in the argument list and make it a required parameter of the function? Otherwise I would rephrase this say instructions instead of circuits and say `two_qubit_gates` so it is clear what is missing.